libsignal-protocol/src/raw_ptr.rs

```diff
@@ -0,0 +1,71 @@
+use std::mem;
+
+/// A wrapper around `libsignal-protocol`'s native reference counted pointers.
+#[derive(Debug)]
+pub struct Raw<T: SignalType>(*mut T);
```

Comment:
So, this `Raw` structure now is our `Rc` interfere with the `libsignal`'s reference counter implementation ?

I didn't know that they allowed to use there implementation for the end users to act as gc or memory management.
I thought it was only used internally :/ .

Reply:
I don't believe the signal's reference counting is an internal thing.

Their [Memory Management][mm] notes explicitly tell people to use the `SIGNAL_REF` and `SIGNAL_UNREF` macros to deal with reference counts. This wrapper just calls those macros (well, technically the upcast to `signal_type_base` and function call it expands to) automatically for us.

> Most of the other data types, including everything internal, use a reference counting mechanism. If you are going to hold onto a reference to one of these types, use the SIGNAL_REF(x) macro to increment its count. If you are done with a reference, use SIGNAL_UNREF(x) to decrement its count. When the count reaches 0, the type's destructor function is called.

[mm]: https://github.com/signalapp/libsignal-protocol-c#memory-management-notes